lib/Paws/Credential/ECSContainerProfile.pm

```diff
@@ -0,0 +1,127 @@
+package Paws::Credential::ECSContainerProfile;
+  use JSON::MaybeXS;
+  use Moose;
+  use DateTime::Format::ISO8601;
+  with 'Paws::Credential';
+
+  my $defaultMetadataUrl = undef;
+
+  my $containerLocalUrl=$ENV{'AWS_CONTAINER_CREDENTIALS_RELATIVE_URI'};
+
+  if ($containerLocalUrl)
+  {
+    $defaultMetadataUrl = "http://169.254.170.2$containerLocalUrl";
```

Comment:
I recommend using the URI module for URL construction like this.

    use URI;
    ...
    $defaultMetadataUrl = URI->new('http://169.254.170.2');
    $defaultMetadataUrl->path($containerLocalUrl);